pkg/engine/renderer/dependencies.go

```diff
@@ -138,16 +139,25 @@ func (de *dependencyCalculator) getHashForDependency(d resourceDependency) (hash
 	if err != nil {
 		return hashBytes{}, fmt.Errorf("failed to get dependeny %s/%s: %v", d.namespace, d.name, err)
 	}
+
+	if dep == nil {
```

Comment:
usually there is an `apierrors.IsNotFound(err)` or the like... how do we know that `dep==nil` is NotFound?
It isn't clear if the message below is correct